Dockerfile

```diff
@@ -0,0 +1,48 @@
+FROM tiangolo/uwsgi-nginx-flask:python2.7
```

Comment:
For us to use this in trusted environments, I think we'll need to start by using a FROM line from at known distro base, like debian, ubuntu-core or alpine.

Reply:
@khilman, the base image *tiangolo/uwsgi-nginx-flask:python2.7* adds flask to the 
 *tiangolo/uwsgi-nginx:python2.7* image, which adds uwsgi and nginx on top of the official *python:2.7* image (python installed on top of a Debian Jessie).
Basically, in each image, some lib and bin are added to build the final image.

Having several processes in the same image is not the recommended approach but I think this can be the first step to Dockerize the frontend.

The next step would be to go one step further and to split the web server (nginx) and the application server (uwsgi).

What do you think ?

Reply:
The issue I have with those images is that I don't know how much I can trust them to always be around or up to date.

Even if they are dead easy to use and includes everything (if not even more) we need, I still prefer to have a base image, based on an official image or out of minideb, and add what is needed (and just what is needed).

My initial idea for building the kernelci Docker images was to have a base image based on minideb with the necessary pieces shared by frontend and backend already installed (git, pip, python...), and then start from that one to build the rest. The final frontend and backend images are going to be big anyway, since we need to install a few libraries to compile the dependencies.

We might not need git though to install the frontend/backend, but we would have to start doing releases here on Github (doable, we just never had a strategy or the need to do it though).

The problem might be with images for nginx and the database layer, but Bitnami provides nginx, Mongodb and Redis images based on their minideb (so they all should share the same base layer).

Reply:
I was thinking about starting from Alpine images for all services.
Not sure about the Bitnami approach, why not using the official images for nginx, mongo and redis ?

Reply:
I would prefer Debian based images, for packaging and historic reasons (and also because if we have to hack around in an image, we don't have to think how the Apline/busybox lay out things, muscle memory).

Bitnami and the minideb image offer a small footprint for a bare minimum Debian based Docker image, with a few utilities to install packages (they have a wrapper script that takes care of cleaning up all the unnecessary cruft left over after installing a package).

The "problem" with the official images is that sometimes they tend to be bigger than necessary, but as long as we don't have to do much with those (apart from config changes) and the image size is comparable, by all means we can use those.

Reply:
ok, got it, no problem to go with the minideb images then.

Regarding the official image, I think it's safe to go with their alpine version though.